src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -429,6 +434,69 @@ def generate_request_payload(configuration, location, public_key, tags):
     return cc
 
 
+def get_kubeconfig_node_dict(kube_config=None):
+    if kube_config is None:
+        kube_config = os.getenv('KUBECONFIG') if os.getenv('KUBECONFIG') else os.path.join(os.path.expanduser('~'), '.kube', 'config')
+    try:
+        kubeconfig_data = config.kube_config._get_kube_config_loader_for_yaml_file(kube_config)._config
+    except Exception as ex:
+        telemetry.set_exception(exception=ex, fault_type=consts.Load_Kubeconfig_Fault_Type,
+                                summary='Error while fetching details from kubeconfig')
+        raise CLIError("Error while fetching details kubeconfig." + str(ex))
+    return kubeconfig_data
+
+
+def check_aks_cluster(kube_config, kube_context):
+    try:
+        config_data = get_kubeconfig_node_dict(kube_config=kube_config)
+    except Exception as e:
+        telemetry.set_user_fault()
+        telemetry.set_exception(exception=e, fault_type=consts.Kubeconfig_Failed_To_Load_Fault_Type,
+                                summary='Problem loading the kubeconfig file')
+        raise CLIError("Problem loading the kubeconfig file: " + str(e))
+
+    try:
+        all_contexts, current_context = config.list_kube_config_contexts(config_file=kube_config)
+    except Exception as e:  # pylint: disable=broad-except
+        telemetry.set_user_fault()
+        telemetry.set_exception(exception=e, fault_type=consts.Load_Kubeconfig_Fault_Type,
+                                summary='Problem listing kube contexts')
+        logger.warning("Exception while trying to list kube contexts: %s\n", e)
+        raise CLIError("Problem listing kube contexts." + str(e))
+
+    if kube_context is None:
+        # Get name of the cluster from current context as kube_context is none.
+        cluster_name = current_context.get('context').get('cluster')
```

Comment:
What happens if the context field is not present in the kubeconfig? We should handle this if it will throw an exception

Reply:
The clierror error will be thrown previously while doing **config.load_kube_config(config_file=kube_config, context=kube_context)**, without this context filled in the kubeconfig.

Reply:
Have you verified it?

Reply:
Yeah, i checked by removing the corresponding context field. load_kube_config throws an error.